src/mixer/playerinfo.cpp

```diff
@@ -129,7 +129,7 @@ void PlayerInfo::updateCurrentPlayingDeck() {
         }
 
         double xfl, xfr;
-        EngineXfader::getXfadeGains(m_pCOxfader->get(), 1.0, 0.0, false, false,
+        EngineXfader::getXfadeGains(m_pCOxfader->get(), 1.0, 0.0, MIXXX_XFADER_ADDITIVE, false,
```

Comment:
This is bug, bt at least this patch does not change the situation. 

Mixxx treats the deck with the highest volume as current track. 
If you enable hamster style for the crossfader, the result is wrong. 

A corner case here is this: A deck is playing at center orientation and two decks ar playing in crossfader center position. With the current implementation all decks have the equal volume and the lowest index is picked. 
The trouth is, that the center deck has the highest volume so this should be picked, but is this the current? 
IMHO yes. 
So for fixing this bug, we need call it with all options like here: 
https://github.com/potocpav/mixxx/blob/const-power-crossfader/src/engine/enginemaster.cpp#L402
The access them we need new ControlObjectSlave obejcts. 

You can fix this issue if you like or place a TODO.